### ISSUE TITLE
olly sometimes fails to create cursor 

### ISSUE THREAD

**Issue by tmcgilchrist** [tarides/runtime_events_tools#33]
```
olly: internal error, uncaught exception:
      Failure("Runtime_events: could not create cursor for specified path.")
      Raised by primitive operation at Dune__exe__Olly.olly in file "bin/olly.ml", line 123, characters 15-71
      Called from Cmdliner_term.app.(fun) in file "cmdliner_term.ml", line 24, characters 19-24
      Called from Cmdliner_eval.run_parser in file "cmdliner_eval.ml", line 34, characters 37-44
```

Raising here for visibility, the error is intermittent and probably caused by a race.

**Comment by tmcgilchrist** [comment]
From what I can tell this issue is caused by MacOS using symlinks for `TMPDIR` and there being a race while opening the temporary file in that location. Switching the call to use `realpath` fixes the issue for me on macOS Sonoma.

``` 
$ echo $TMPDIR
/var/folders/zc/68pdnwdj0cj89c31j5rhjq2c0000gp/T/
$ realpath $TMPDIR  
/private/var/folders/zc/68pdnwdj0cj89c31j5rhjq2c0000gp/T
```

**Comment by tmcgilchrist** [comment]
Fixed and released in 0.5.1
